libvast/include/vast/system/catalog.hpp

```diff
@@ -73,6 +75,24 @@ struct catalog_state {
   /// Update the list of fields that should not be touched by the pruner.
   void update_unprunable_fields(const partition_synopsis& ps);
 
+  /// Summarizes the actors state.
+  [[nodiscard]] record status(status_verbosity v) const;
+
+  /// Create the path that the catalog's type registry is persisted at on disk.
+  [[nodiscard]] std::filesystem::path type_registry_filename() const;
+
+  /// Save the type-registry to disk.
+  [[nodiscard]] caf::error save_to_disk() const;
+
+  /// Load the type-registry from disk.
+  caf::error load_from_disk();
```

Comment:
Should this be `{save,load}_type_registry_{to,from}_disk()` to be very explicit?

Reply:
OK - added in https://github.com/tenzir/vast/pull/2745/commits/df0875de9b419102a0ba09136b8649bd791a5af2